data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultReactiveCosmosRepositoryOperations.java

```diff
@@ -276,7 +278,9 @@ public <T> Mono<Long> count(PagedQuery<T> pagedQuery) {
     @NonNull
     public <T, R> Flux<R> findAll(@NonNull PreparedQuery<T, R> pq) {
         SqlPreparedQuery<T, R> preparedQuery = getSqlPreparedQuery(pq);
-        preparedQuery.attachPageable(preparedQuery.getPageable(), preparedQuery.getQueryLimit(), preparedQuery.getSort(), false);
+        Pageable pageable = preparedQuery.getPageable();
+        Limit queryLimit = Limit.of(pageable.getSize(), pageable.getOffset());
```

Comment:
paging test in azure cosmos was failing, I just tried with latest version to do sanity check.
Not sure why `preparedQuery.getQueryLimit()` returned `Limit.UNLIMITED` but this workaround fixes it for azure cosmos queries.